File: contains_duplicate/contains_duplicate.js
// Copyright 2022, Adewumi Sunkanmi
/**
 * @param {number[]} nums
 * @return {boolean}
 */
 var containsDuplicate = function(nums) {
    if(nums.length==0){
        return false
    }
    const map ={}
    for(let i of nums){
        if(!map[i]){
            map[i] = true
        }else{
           return true
        }
        
    }
    
    return false
};

File: contains_duplicate/contains_duplicate.go
package containsduplicate

func containsDuplicate(nums []int) bool {

	countOccurenceMap := map[int]bool{}
	for i := range nums {
		if countOccurenceMap[nums[i]] {
			return true
		} else {
			countOccurenceMap[nums[i]] = true
		}
	}
	return false
}
